frontend/src/pages/RoulettePage/RoulettePage.tsx

```diff
@@ -19,18 +19,22 @@ const RoulettePage = () => {
 
   const isRouletteView = currentView === 'roulette';
 
+  const handleViewChange = () => {
+    setCurrentView((prev) => (prev === 'roulette' ? 'statistics' : 'roulette'));
+  };
+
   const handleSpinClick = () => {
+    if (currentView === 'statistics') {
+      handleViewChange();
+    }
+
     setSpinning(true);
     setTimeout(() => {
       setSpinning(false);
       navigate('/room/:roomId/roulette/result');
     }, 3000);
   };
```

Comment:
여기 setTimeout 부분에는 사이드 이펙트가 존재하니까 따로 useEffect에서 실행해주는 건 어때용? 클릭처리하는 부분이랑 사이드 이펙트 부분을 따로 분리해두면 훨씬 책임분리도 잘 되고 가독성 좋아질 것 같아용!

그리고 clearTimeout 해서 메모리 누수도 막아주면 좋을 것 같습니다!

```suggestion
  const handleViewChange = () => {
    setCurrentView((prev) => (prev === 'roulette' ? 'statistics' : 'roulette'));
  };

  const handleSpinClick = () => {
    if (currentView === 'statistics') handleViewChange();
    setSpinning(true);
  };

  useEffect(() => {
    if (spinning) {
      const timer = setTimeout(() => {
        setSpinning(false);
        navigate(`/room/:roomId/roulette/result`);
      }, 3000);
      return () => clearTimeout(timer);
    }
  }, [navigate, spinning]);
```